frontend/src/DialectReportView.tsx

```diff
@@ -84,23 +89,16 @@ export const DialectReportView = ({
 const filterOtherImplementations = (
   allImplementationsData: Record<string, Implementation>,
   langs: string[],
-  filteredReportImplementationsMap: Map<string, ImplementationData>,
+  filteredReportImplementationsMap: Map<string, ImplementationData>
 ): Record<string, Implementation> => {
   const filteredOtherImplementationsArray: [string, Implementation][] =
     Object.entries(allImplementationsData).filter(
-      ([, impl]: [string, Implementation]) => langs.includes(impl.language),
+      ([, impl]: [string, Implementation]) => langs.includes(impl.language)
     );
 
   const filteredOtherImplementationsData: Record<string, Implementation> = {};
-  filteredOtherImplementationsArray.forEach(([key, impl]) => {
-    let isFiltered = false;
-    for (const val of filteredReportImplementationsMap.values()) {
-      if (impl.source === val.metadata.source) {
-        isFiltered = true;
-        break;
-      }
-    }
-    if (!isFiltered) {
+  filteredOtherImplementationsArray.map(([key, impl]) => {
```

Comment:
Maybe I mislead you a bit with `map()` - but you are using it here incorrectly, mapping logic should be side effect free and should return transformed/mapped value for each element.

Actually, just filtering is enough:
```ts
  const filteredOtherImplementationsArray = Object.entries(allImplementationsData)
    .filter(([, impl]) => langs.includes(impl.language),)
    .filter(([key]) => !filteredReportImplementationsMap.has(key));
  return Object.fromEntries(filteredOtherImplementationsArray);
```
I believe this should do the same thing and is more concise. But please ensure if I didn't miss anything :)

PS: IMO annotating everything with TS types is just making things less readable. As a rule of thumb you shouldn't write any types as long as type inference works correctly. In the case above I just get rid of all the annotations and the type information is not lost and you're getting all the benefits the same way you were getting before